usr.sbin/bhyve/migration.c

```diff
@@ -934,40 +764,35 @@ migrate_devs(struct vmctx *ctx, int socket, enum migration_transfer_req req)
 		 * be migrated.
 		 */
 		snapshot_devs = get_snapshot_devs(&num_items);
-		rc = migration_send_data_remote(socket, &num_items, sizeof(num_items));
 
+		rc = migration_transfer_data(socket, &num_items, sizeof(num_items), req);
 		if (rc < 0) {
-		    fprintf(stderr, "%s: Could not send num_items to destination\r\n", __func__);
-		    return (-1);
+			DPRINTF("Could not send num_items to destination");
+			return (-1);
 		}
 
 		for (i = 0; i < num_items; i++) {
-			rc = migrate_send_dev(ctx, socket, snapshot_devs[i].dev_name,
-						buffer, SNAPSHOT_BUFFER_SIZE);
+			rc = migrate_transfer_dev(ctx, socket, snapshot_devs[i].dev_name,
+						buffer, SNAPSHOT_BUFFER_SIZE, req);
 
 			if (rc < 0) {
-				fprintf(stderr,
-					"%s: Could not send %s\r\n",
-					__func__, snapshot_devs[i].dev_name);
+				DPRINTF("Could not send %s", snapshot_devs[i].dev_name);
 				error = -1;
 				goto end;
 			}
 	    }
 	} else if (req == MIGRATION_RECV_REQ) {
```

Comment:
For similar reasons as the kernel structures, the conditionals here could be merged, as they perform the same operations (almost entirely). Could probably add a check to see if the other end sends the expected device, or they are out of sync (possibly different versions of bhyve installed?).